src/Uno.UI.Composition/Composition/CompositionClip.skia.cs

```diff
@@ -10,8 +11,23 @@ partial class CompositionClip
 {
 	/// <summary>
 	/// Returns the bounds of the clip. The clip itself could be non-rectangular, e.g, rounded rectangle or path.
+	/// Note that this already handles TransformMatrix
 	/// </summary>
-	internal virtual Rect? GetBounds(Visual visual)
+	internal Rect? GetBounds(Visual visual)
+	{
+		if (GetBoundsCore(visual) is { } bounds)
+		{
+			return bounds.Transform(this.TransformMatrix);
```

Comment:
I would do the opposite actually, transformation are computed using matrix, the `rect.Transform` is an extensions method kept for backward compatibility:

``` suggestion
			return TransformMatrix.Transform(rect);
```